src/test/java/com/epam/tc/hw2/ex1/Exercise1Test.java

```diff
@@ -0,0 +1,131 @@
+        WebElement userFirstSecondName = driver.findElement(By.id("user-name"));
+
+        // wait until the username is no more hidden
+        webDriverWait.until(ExpectedConditions.visibilityOf(userFirstSecondName));
+
+        softAssertions.assertThat(userFirstSecondName.getText()).isEqualTo(usernameInfo);
+
+        // 5) Assert that there are 4 items on the header -------------------------------------
+        // section are displayed and they have proper texts
+        List<WebElement> navBarItems = driver.findElements(By.xpath("//nav/ul/li/a[not(@href='#')]"));
+        assertItemsCountPlacementAndTexts(navBarItems, 4, expectedNavBarTitles, softAssertions);
+
+        // 6) Assert that there are 4 images on the
+        // Index Page and they are displayed
+        List<WebElement> icons = driver.findElements(By.className("icons-benefit"));
+        softAssertions.assertThat(icons.size()).isEqualTo(4);
+        icons.forEach(icon -> softAssertions.assertThat(icon.isDisplayed()).isTrue());
+
+        // 7) Assert that there are 4 texts on the
+        // Index Page under icons and they have proper text
+        List<WebElement> textsUnderImages = driver.findElements(By.className("benefit-txt"));
+        assertItemsCountPlacementAndTexts(textsUnderImages, 4, expectedTextsUnderIcons, softAssertions);
+
+        // 8) Assert that there is the iframe with “Frame Button” exist
+        WebElement frame = driver.findElement(By.id("frame"));
+        softAssertions.assertThat(frame).isNotNull();
+
+        // 9) Switch to the iframe and check that there is “Frame Button” in the iframe
+
+        // remembering of the main window
+        String windowHandle = driver.getWindowHandle();
+
+        driver.switchTo().frame("frame");
+        WebElement frameButton = driver.findElement(By.id("frame-button"));
+        softAssertions.assertThat(frameButton).isNotNull();
+
+        // 10) Switch to original window back
+        driver.switchTo().window(windowHandle);
+
+        // 11) Assert that there are 5 items in the Left Section
+        // are displayed and they have proper text
+        List<WebElement> actualLeftItems = driver.findElements(By.cssSelector(".wrapper ul.left >li > a > span"));
+        assertItemsCountPlacementAndTexts(actualLeftItems, 5, expectedMenuTitles, softAssertions);
+
+        softAssertions.assertAll();
+    }
+
+    public void assertItemsCountPlacementAndTexts(List<WebElement> actualItems, int expectedCount,
+                                                  String[] expectedTexts, SoftAssertions softAssertions) {
+        // assert count
+        softAssertions.assertThat(actualItems.size()).isEqualTo(expectedCount);
+
+        // assert placement
+        actualItems.forEach(element -> softAssertions.assertThat(element.isDisplayed()).isTrue());
+
+        // assert texts
+        List<WebElement> actualItemsTexts = new ArrayList<>(actualItems);
```

Comment:
Looks like `actualItemsTexts` is just copying `actualItems`...

Reply:
Changed `actualItemsTexts` type to String and now it contains only texts of the elements.